syntax = "proto3";

package payment_service;


service PaymentService { 
    rpc ProcessPayment(PaymentRequest) returns (PaymentResponse); 
    }


message PaymentRequest { 
    string orderId = 1; 
    double amount = 2; 
    string currency = 3;
    bool commit = 4;
    }

message PaymentResponse { 
    string orderId = 1;
    bool success = 2; 
    string message = 3; 
    }